### PR TITLE
Pull #12154: Use OS specific command and correct errorExtractingPattern

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#### 1. Usage of OS specific command: (Exception without this modification in Windows OS) (Bug Fix)
```
Caught: java.io.IOException: Cannot run program "mvn": CreateProcess error=2, The system cannot find the file specified
java.io.IOException: Cannot run program "mvn": CreateProcess error=2, The system cannot find the file specified
        at error-prone-check.getErrorProneErrors(error-prone-check.groovy:87)
        at error-prone-check.checkErrorProneReport(error-prone-check.groovy:63)
        at error-prone-check.parseArgumentAndExecute(error-prone-check.groovy:41)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at error-prone-check.run(error-prone-check.groovy:19)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
        ... 10 more
```
#### 3. Removed redundant escapes (`\\]`) (Enhancement)

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
Works fine, tested locally on windows.

**Comment by Vyom-Yadav** [comment]
`
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
/home/runner/work/checkstyle/checkstyle/.ci/error-prone-check.groovy: 12: Apparent variable 'SEPARATOR' was found in a static scope but doesn't refer to a local variable, static field or class. Possible causes:
`
There were a lot of bugs fixed related to static fields in `2.5` (See [this](http://groovy-lang.org/changelogs/changelog-2.5.0.html#changelog)). This works correctly in `2.5` but we are using `2.4`, I think we should consider upgrading somehow.

File: .ci/error-prone-check.groovy [hunk 1]
```diff
@@ -8,7 +8,8 @@ import java.util.regex.Pattern
 
 @Field static final String SEPARATOR = System.getProperty("file.separator")
```
Comment (nrmancuso) [inline .ci/error-prone-check.groovy:9]: Why aren't we using `File.separator` ?

File: .ci/error-prone-check.groovy [hunk 2]
```diff
@@ -82,7 +83,14 @@ private static int checkErrorProneReport(String profile, String flag) {
  */
 private static List<String> getErrorProneErrors(String profile) {
     final List<String> errorProneErrors = [] as ArrayList
-    final String command = "mvn -e --no-transfer-progress clean" +
+    final String maven
+    if ("\\".equals(SEPARATOR)) {
+        maven = "mvn.cmd"
```
Comment (rnveach) [inline .ci/error-prone-check.groovy:88]: Why?
https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/diff.groovy#L219

Comment (Vyom-Yadav) [reply .ci/error-prone-check.groovy:88]: That is a user-defined method:
```groovy
def executeCmd(cmd, dir = new File("").absoluteFile) {
    println "Running command: ${cmd}"
    def osSpecificCmd = getOsSpecificCmd(cmd)
    def proc = osSpecificCmd.execute(null, dir)
    proc.consumeProcessOutput(System.out, System.err)
    proc.waitFor()
    if (proc.exitValue() != 0) {
        throw new GroovyRuntimeException("Error: ${proc.err.text}!")
    }
}
```

File: .ci/error-prone-check.groovy (hunk 1 above)
Comment (Vyom-Yadav) [reply .ci/error-prone-check.groovy]: Done.

File: .ci/error-prone-check.groovy (hunk 2 above)
Comment (rnveach) [reply .ci/error-prone-check.groovy]: Yes, but it doesn't need to swap from running `mvn` to `mvn.cmd`. Also, I executed this on Windows and it ran fine without the `.cmd`.

Comment (Vyom-Yadav) [reply .ci/error-prone-check.groovy]: I got `java.io.IOException: Cannot run program "mvn": CreateProcess error=2, The system cannot find the file specified`, are the current changes fine?

**Comment by romani** [comment]
please change commit to `Pull #12154: .....` and explain kin PR description reason on why this update is required.
I am curious why were  ok without in past and who will benefit from this update, and what problem had such people wihtout this update.

**Review by nrmancuso — APPROVED** [review]
I am good as long as CI is passing and we have confirmation from @rnveach that this script works. I do not have access to a windows machine, and I do not think it is worth it to add to Windows CI just to prove it.

**Comment by rnveach** [comment]
````
Summary: TotalFiles=3 FilesWithViolations=1 P1=0 P2=0 P3=1

File: error-prone-check.groovy
    Violation: Rule=UnnecessarySemicolon P=3 Line=124 Msg=[Semi-colons as line endings can be removed safely] Src=[            .compile(".*[\\\\/](.*\\.java):\\[(\\d+).*\\[(\\w+)](.*)");]

[CodeNarc (http://www.codenarc.org/) v1.4]
CodeNarc completed: (p1=0; p2=0; p3=1) 5768ms
````

File: .ci/error-prone-check.groovy (hunk 2 above)
Comment (rnveach) [reply .ci/error-prone-check.groovy]: I am good with the new changes and am resolving this.

**Review by rnveach — CHANGES_REQUESTED** [review]
CI failure.

**Comment by Vyom-Yadav** [comment]
> CI failure.

Fixed

**Comment by romani** [comment]
>Usage of OS specific command:

Please share a command and what OS you use.
I still do not understand what problem we have.we never had such problem. In what condition it happens?

**Comment by Vyom-Yadav** [comment]
> > Usage of OS specific command:
> 
> Please share a command and what OS you use. I still do not understand what problem we have.we never had such problem. In what condition it happens?

Only this script uses maven. We never had this problem because the contribution repo is handling it properly. See https://github.com/checkstyle/contribution/blob/1b3e33600898a849791dc88c90df2bb3f80400e7/checkstyle-tester/diff.groovy#L669-L689

You can normally execute `mvn clean verify` in windows, but when you use it like `command.execute()` then it throws and exception.

**Comment by Vyom-Yadav** [comment]
See https://stackoverflow.com/q/9674220/14425335

**Comment by romani** [comment]
Ok it is win os, for new script that we created recently.

**Review by rnveach — APPROVED** [review]

**Pull request merged**